### PR TITLE
feat: api update personal info

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] update employee's personal api

#### What are the relevant Git tickets?
https://www.notion.so/dwarves/API-to-edit-employee-info-62f923ea04a843ad96794cd1d0d3cb50

#### Screenshots (if appropriate)
image

#### Request
```
curl 'http://localhost:8200/api/v1/employees/2655832e-f009-4b73-a535-64c3a22e558f/personal-info'
-X PUT
-H 'accept: /'
-H 'accept-language: vi,en-US;q=0.9,en;q=0.8'
-H 'content-type: application/json'
-H 'origin: http://localhost:3000/'
-H 'referer: http://localhost:3000/'
-H 'sec-ch-ua: "Chromium";v="106", "Google Chrome";v="106", "Not;A=Brand";v="99"'
-H 'sec-ch-ua-mobile: ?0'
-H 'sec-ch-ua-platform: "macOS"'
-H 'sec-fetch-dest: empty'
-H 'sec-fetch-mode: cors'
-H 'sec-fetch-site: cross-site'
-H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36'
-H 'authorization:Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTkzMjExNDIsImlkIjoiMjY1NTgzMmUtZjAwOS00YjczLWE1MzUtNjRjM2EyMmU1NThmIiwiYXZhdGFyIjoiaHR0cHM6Ly9zMy1hcC1zb3V0aGVhc3QtMS5hbWF6b25hd3MuY29tL2ZvcnRyZXNzLWltYWdlcy81MTUzNTc0Njk1NjYzOTU1OTQ0LnBuZyIsImVtYWlsIjoidGhhbmhAZC5mb3VuZGF0aW9uIiwicGVybWlzc2lvbnMiOlsiZW1wbG95ZWVzLnJlYWQiXSwidXNlcl9pbmZvIjpudWxsfQ.GENGPEucSUrILN6tHDKxLMtj0M0REVMUPC7-XhDMpGM'
--data-raw '{
    "dob":"1990-01-02T00:00:00Z",
    "gender":"Male",
    "address":"Phan Huy Ich, Tan Binh District, Ho Chi Minh, Vietnam",
    "personalEmail":"thanhpham123@gmail.com"
}'
--compressed
```
#### Response
Update general info
```
{
    "data": {
        "id": "2655832e-f009-4b73-a535-64c3a22e558f",
        "createdAt": "2022-11-02T09:52:34.586566Z",
        "updatedAt": "2022-11-16T10:09:01.798162Z",
        "fullName": "Phạm Đức Thành",
        "teamEmail": "thanh@d.foundation",
        "phoneNumber": "0123456788",
        "discordID": "",
        "githubID": "",
        "notionID": "",
        "lineManager": {
            "id": "ecea9d15-05ba-4a4e-9787-54210e3b98ce",
            "fullName": "Nguyễn Hoàng Huy",
            "avatar": "https://s3-ap-southeast-1.amazonaws.com/fortress-images/2830497479497502617.png"
        }
    }
}
```

Update skills
```
{
    "data": {
        "id": "2655832e-f009-4b73-a535-64c3a22e558f",
        "createdAt": "2022-11-02T09:52:34.586566Z",
        "updatedAt": "2022-11-16T10:00:30.613898Z",
        "seniority": {
            "id": "39735742-829b-47f3-8f9d-daf0983914e5",
            "createdAt": "2022-11-07T09:50:25.714604Z",
            "updatedAt": "2022-11-07T09:50:25.714604Z",
            "name": "Principal",
            "code": "principal"
        },
        "chapter": {
            "id": "11ccffea-2cc9-4e98-9bef-3464dfe4dec8",
            "createdAt": "2022-11-07T09:50:25.714604Z",
            "updatedAt": "2022-11-07T09:50:25.714604Z",
            "name": "Web",
            "code": "web"
        },
        "positions": [
            {
                "id": "11ccffea-2cc9-4e98-9bef-3464dfe4dec8",
                "createdAt": "2022-11-07T09:50:25.714604Z",
                "updatedAt": "2022-11-07T09:50:25.714604Z",
                "name": "Frontend",
                "code": "frontend"
            },
            {
                "id": "d796884d-a8c4-4525-81e7-54a3b6099eac",
                "createdAt": "2022-11-07T09:50:25.714604Z",
                "updatedAt": "2022-11-07T09:50:25.714604Z",
                "name": "Backend",
                "code": "backend"
            }
        ],
        "stacks": [
            {
                "id": "0ecf47c8-cca4-4c30-94bb-054b1124c44f",
                "createdAt": "2022-11-11T18:38:46.266725Z",
                "updatedAt": "2022-11-11T18:38:46.266725Z",
                "name": "Golang",
                "code": "golang"
            },
            {
                "id": "fa0f4e46-7eab-4e5c-9d31-30489e69fe2e",
                "createdAt": "2022-11-11T18:38:46.266725Z",
                "updatedAt": "2022-11-11T18:38:46.266725Z",
                "name": "React",
                "code": "react"
            }
        ]
    }
}
```

Update Personal Info
```
{
    "data": {
        "id": "2655832e-f009-4b73-a535-64c3a22e558f",
        "createdAt": "2022-11-02T09:52:34.586566Z",
        "updatedAt": "2022-11-16T09:52:30.359844Z",
        "personalEmail": "thanhpham123@gmail.com",
        "address": "Phan Huy Ich, Tan Binh District, Ho Chi Minh, Vietnam",
        "gender": "Male",
        "birthday": "1990-01-02T00:00:00Z"
    }
}
```

### CODE REVIEW

File: pkg/handler/employee/employee.go [hunk 1]
```diff
@@ -294,10 +288,17 @@ func (h *handler) UpdateGeneralInfo(c *gin.Context) {
 	}, employeeID)
 
 	if err != nil {
+		if err == gorm.ErrRecordNotFound {
```
Comment (namnhce) [inline pkg/handler/employee/employee.go:291]:
```suggestion
		if errors.Is(err, ErrRecordNotFound) {
```

File: pkg/handler/employee/employee.go [hunk 2]
```diff
@@ -458,15 +452,70 @@ func (h *handler) UpdateSkills(c *gin.Context) {
 
 	if err != nil {
 		if err == gorm.ErrRecordNotFound {
```
Comment (namnhce) [inline pkg/handler/employee/employee.go:454]:
```suggestion
		if errors.Is(err, ErrRecordNotFound) {
```

File: pkg/handler/employee/employee.go [hunk 3]
```diff
@@ -458,15 +452,70 @@ func (h *handler) UpdateSkills(c *gin.Context) {
...
+	// 3. update informations and rerurn
+	rs, err := h.store.Employee.UpdatePersonalInfo(h.repo.DB(), employee.UpdatePersonalInfoInput{
+		DoB:           body.DoB,
+		Gender:        body.Gender,
+		Address:       body.Address,
+		PersonalEmail: body.PersonalEmail,
+	}, employeeID)
+
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
```
Comment (namnhce) [inline pkg/handler/employee/employee.go:509]:
```suggestion
		if errors.Is(err, ErrRecordNotFound) {
```

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/handler/employee/employee.go (hunk 1 above)
Comment (datphamcode295) [reply pkg/handler/employee/employee.go]: updated

File: pkg/handler/employee/employee.go (hunk 3 above)
Comment (datphamcode295) [reply pkg/handler/employee/employee.go]: updated

File: pkg/handler/employee/employee.go (hunk 2 above)
Comment (datphamcode295) [reply pkg/handler/employee/employee.go]: updated

**Review by namnhce — APPROVED** [review]

**Pull request merged**